packages/orchestration/test/exos/chain-hub.test.ts

```diff
@@ -191,3 +203,371 @@ test('makeChainAddress', async t => {
+  ]);
+});
+
+const knownChainsSansConns = objectMap(
+  withChainCapabilities(knownChains),
+  ({ connections, ...rest }) => rest,
+);
+
+test('getTransferRoute - no connection info single hop', t => {
+  const { chainHub, inspectLogs } = setup(t);
+
+  registerChainsAndAssets(
+    chainHub,
+    {},
+    knownChainsSansConns, // omit connections
+    harden({ [uusdcOnAgoric]: agDetail }),
+  );
+
+  t.is(
+    chainHub.getTransferRoute(
+      nobleDest,
+      harden({ denom: uusdcOnAgoric, value: 100n }),
+      'agoric',
+    ),
+    undefined,
+  );
+  t.deepEqual(inspectLogs(-1), [
+    'no connection info found for agoric-3_noble-1',
+  ]);
+});
+
+test('getTransferRoute - no connection info multi hop', t => {
+  const { chainHub, inspectLogs } = setup(t);
+
+  // only agoric has connection info; osmosis<>noble will be missing
+  const chainInfo = { ...knownChainsSansConns, agoric: knownChains.agoric };
+  registerChainsAndAssets(
+    chainHub,
+    {},
+    harden(chainInfo),
+    harden({ [uusdcOnAgoric]: agDetail, [uusdcOnOsmosis]: osDetail }),
+  );
+
+  const osmoDest = chainHub.makeChainAddress('osmo1234');
+  const agoricDest = chainHub.makeChainAddress('agoric1234');
+
+  t.is(
+    chainHub.getTransferRoute(
+      osmoDest,
+      harden({ denom: uusdcOnAgoric, value: 100n }),
+      'agoric',
+    ),
+    undefined,
+  );
+  t.deepEqual(inspectLogs(-1), [
+    'no connection info found for noble-1_osmosis-1',
+  ]);
+
+  // transfer USDC on osmosis to agoric
+  t.is(
+    chainHub.getTransferRoute(
+      agoricDest,
+      harden({ denom: uusdcOnOsmosis, value: 100n }),
+      'osmosis',
+    ),
+    undefined,
+  );
+  t.deepEqual(inspectLogs(-1), [
+    'no connection info found for noble-1_osmosis-1',
+  ]);
+});
+
+test('getTransferRoute - asset not on holding chain', t => {
+  const { chainHub, inspectLogs } = setup(t);
+
+  registerChainsAndAssets(
+    chainHub,
+    {},
+    withChainCapabilities(knownChains),
+    harden({ [uusdcOnAgoric]: agDetail }),
+  );
+
+  // transfer USDC on agoric from osmosis to noble (impossible)
+  t.is(
+    chainHub.getTransferRoute(
+      nobleDest,
+      harden({ denom: uusdcOnAgoric, value: 100n }),
+      'osmosis',
+    ),
+    undefined,
+  );
+  t.deepEqual(inspectLogs(-1), [
+    'cannot transfer asset ibc/FE98AAD68F02F03565E9FA39A5E627946699B2B07115889ED812D8BA639576A9. held on agoric not osmosis.',
+  ]);
+});
+
+test('getTransferRoute - no PFM path', t => {
+  const { chainHub, inspectLogs } = setup(t);
+
+  registerChainsAndAssets(
+    chainHub,
+    {},
+    knownChains, // intentionally omit pfmEnabled
+    harden({ [uusdcOnAgoric]: agDetail }),
+  );
+
+  // transfer USDC on agoric to osmosis
+  t.is(
+    chainHub.getTransferRoute(
+      chainHub.makeChainAddress('osmo1234'),
+      harden({ denom: uusdcOnAgoric, value: 100n }),
+      'agoric',
+    ),
+    undefined,
+  );
+  t.deepEqual(inspectLogs(-1), ['pfm not enabled on issuing chain: noble']);
```

Comment:
One downside of the  "return undefined instead of error" approach is that the error info will only be printed to the console. I wonder if it's better to throw instead, so the info is bubbled up to the transfer vow rejection.

Reply:
See [d7b3e96](https://github.com/Agoric/agoric-sdk/pull/10584/commits/d7b3e966cf6400eadb5cf59b3ddc3a93e8ad76ce) where this now throws so errors can bubble up